contrib/mesos/pkg/minion/server.go

```diff
@@ -206,17 +206,23 @@ func (ms *MinionServer) launchHyperkubeServer(server string, args []string, logF
 		}
 	}
 
-	// use given environment, but add /usr/sbin to the path for the iptables binary used in kube-proxy
+	// use given environment, but add /usr/sbin and $SANDBOX/bin to the path for the iptables binary used in kube-proxy
 	var kmEnv []string
-	if ms.pathOverride != "" {
-		env := os.Environ()
-		kmEnv = make([]string, 0, len(env))
-		for _, e := range env {
-			if !strings.HasPrefix(e, "PATH=") {
-				kmEnv = append(kmEnv, e)
+	env := os.Environ()
+	kmEnv = make([]string, 0, len(env))
+	for _, e := range env {
+		if !strings.HasPrefix(e, "PATH=") {
+			kmEnv = append(kmEnv, e)
+		} else {
+			if ms.pathOverride != "" {
```

Comment:
you're right. i should avoid doing code reviews when I'm so tired.

On Sat, Oct 24, 2015 at 2:10 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/minion/server.go
> <https://github.com/kubernetes/kubernetes/pull/16111#discussion_r42931004>
> :
>
> >  	var kmEnv []string
> > -	if ms.pathOverride != "" {
> > -		env := os.Environ()
> > -		kmEnv = make([]string, 0, len(env))
> > -		for _, e := range env {
> > -			if !strings.HasPrefix(e, "PATH=") {
> > -				kmEnv = append(kmEnv, e)
> > +	env := os.Environ()
> > +	kmEnv = make([]string, 0, len(env))
> > +	for _, e := range env {
> > +		if !strings.HasPrefix(e, "PATH=") {
> > +			kmEnv = append(kmEnv, e)
> > +		} else {
> > +			if ms.pathOverride != "" {
>
> The else-block only executes for variables which starts with PATH=.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16111/files#r42931004>.
>